From 1a040201db32119759e50bfcbbd620f1c7f205e4 Mon Sep 17 00:00:00 2001
From: "sheng.liu" <sheng.liu@amlogic.com>
Date: Sun, 26 Sep 2021 10:42:36 +0800
Subject: [PATCH] gstreamer1.0-plugins-good : qtdemux send data by dts [1/1]

PD#SWPL-60007

Problem:
qtdemux switch to video stream or audio stream by dts to send data,
which is not correct, because render data is by pts.

Solution:
Use pts instead of dts to switch video/audio stream

Verify:
AP222

Change-Id: I3a9cefe57b30be889bc2b44fbe56b75eab821702
---
 gst/isomp4/qtdemux.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index d438682..c6724aa 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -5337,14 +5337,14 @@ gst_qtdemux_advance_sample (GstQTDemux * qtdemux, QtDemuxStream * stream)
   sample = &stream->samples[stream->sample_index];
 
   /* see if we are past the segment */
-  if (G_UNLIKELY (QTSAMPLE_DTS (stream, sample) >= segment->media_stop))
+  if (G_UNLIKELY (QTSAMPLE_PTS (stream, sample) >= segment->media_stop))
     goto next_segment;
 
-  if (QTSAMPLE_DTS (stream, sample) >= segment->media_start) {
+  if (QTSAMPLE_PTS (stream, sample) >= segment->media_start) {
     /* inside the segment, update time_position, looks very familiar to
      * GStreamer segments, doesn't it? */
     stream->time_position =
-        QTSAMPLE_DTS (stream, sample) - segment->media_start + segment->time;
+        QTSAMPLE_PTS (stream, sample) - segment->media_start + segment->time;
   } else {
     /* not yet in segment, time does not yet increment. This means
      * that we are still prerolling keyframes to the decoder so it can
-- 
2.32.0

